### PR TITLE
Timezones: Build auto-suggestion from all known timezones.

### PR DESCRIPTION
**Pull request by siedentop**
Proof of concept for #9.

```Shell
chris@christophs-air rizzy % cargo run -- -z America/Chi
    Finished dev [unoptimized + debuginfo] target(s) in 0.07s
     Running `target/debug/rizzy -z America/Chi`
Error: 
   0: Unknown timezone 'America/Chi', did you mean: ["America/Chihuahua", "America/Chicago"]

Location:
   src/main.rs:103

Backtrace omitted.
Run with RUST_BACKTRACE=1 environment variable to display it.
Run with RUST_BACKTRACE=full to include source snippets.
```

### CODE REVIEW

**Comment by siedentop** [comment]
@mattgodbolt I just found that the `chrono-tz` crate already has a way to get the names of all timezones. This means, there's no upstream changes necessary to make this feature work.

IMO, this is ready to merge. For unrelated reasons, I won't be able to work on it from tomorrow onwards. Please feel free to merge, modify or close as suits your project best. Many thanks for letting me work on rizzy a bit!

**Comment by mattgodbolt** [comment]
Sorry for the slowness, I'll take a look!

File: src/main.rs [hunk 1]
```diff
@@ -105,23 +68,30 @@ where
...
         }
     } else {
         for file in opts.file {
-            if let Ok(lines) = read_lines(&file) {
-                for line in lines {
-                    println!("{}", rizzy.handle_line(&line.unwrap()));
-                }
-            } else {
-                panic!("Failed to open file '{}'", file);
+            let lines =
```
Comment (mattgodbolt) [inline src/main.rs:82]: TIL!

File: src/timezones.rs [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    let suggestions = dictionary.auto_suggest_alternative_words(time_zone);
+    suggestions.unwrap_or_default()
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_correct_spelling() {
```
Comment (mattgodbolt) [inline src/timezones.rs:28]: love it, thanks

File: tests/cli.rs [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+    let stderr = String::from_utf8(assert.get_output().stderr.clone()).unwrap();
+    insta::assert_display_snapshot!(stderr);
+
+    assert
+        .failure()
+        .stderr(predicate::str::contains("America/New_York"));
+}
+
+/// Make sure the examples from the README work.
```
Comment (mattgodbolt) [inline tests/cli.rs:20]: Briliant!

File: tests/cli.rs [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+fn test_from_readme() {
+    let assert = Command::cargo_bin("rizzy")
+        .unwrap()
+        .arg("--chi")
+        .arg("tests/some.log.file")
+        .assert();
+
+    let assert = assert.success().stderr(predicate::eq(""));
+    let stdout = String::from_utf8(assert.get_output().stdout.clone()).unwrap();
+    insta::assert_display_snapshot!(stdout);
```
Comment (mattgodbolt) [inline tests/cli.rs:31]: Took me a bit to realise this `insta` refers to those `.snap` files. But now I get it!

**Review by mattgodbolt — APPROVED** [review]
LGTM! thanks!

**Pull request merged**